File: mslibrary-kotlindocs/src/main/java/kr/msleague/message/adapter/PlayerAdapter.java
package kr.msleague.message.adapter;

import kr.msleague.message.api.MSMessageAdapter;
import org.bukkit.entity.Player;

/**
 * MSLibrary 에서 기본적으로 제공하는 Player 어댑터입니다.
 */
public class PlayerAdapter implements MSMessageAdapter<Player> {
    @Override
    public String reformat(String origin, Player obj) {
        return "";
    }
}

File: mslibrary-message/src/main/java/kr/msleague/message/adapter/PlayerAdapter.kt
package kr.msleague.message.adapter

import kr.msleague.util.extensions.roundString
import kr.msleague.message.api.MSMessageAdapter
import org.bukkit.entity.Player

class PlayerAdapter: MSMessageAdapter<Player> {

    override fun request(obj: Player, placeHolder: String): String? {
        return when(placeHolder) {
            "player_name" -> obj.name
            "player_health" -> obj.health.roundString()
            "player_max_health" -> obj.health.roundString()
            else -> null
        }
    }
}